cachi2/core/package_managers/yarn/resolver.py

```diff
@@ -344,13 +346,45 @@ def _generate_purl_for_package(package: _ResolvedPackage, project: Project) -> s
     :param project: the project object to resolve the configured registry url and file paths
         for file dependencies.
     """
-    # registry url can be accessed in project.yarnrc
-    # paths for file dependencies are relative to project.source_dir
-    return PackageURL(
+    qualifiers = dict()
+    if package.checksum is not None:
+        qualifiers["checksum"] = package.checksum
+
+    subpath = None
+
+    if isinstance(package.locator, HttpsLocator):
+        qualifiers["download_url"] = package.locator.url
+
+    elif isinstance(package.locator, WorkspaceLocator):
+        workspace_path = package.locator.relpath
+        workspace_abs_path = project.source_dir.join_within_root(workspace_path)
+
+        qualifiers["vcs_url"] = str(workspace_abs_path)
+        subpath = str(workspace_abs_path.subpath_from_root)
+
+    elif isinstance(package.locator, (FileLocator, LinkLocator, PortalLocator)):
+        workspace_path = package.locator.locator.relpath
+        workspace_abs_path = project.source_dir.join_within_root(workspace_path)
+        package_abs_path = workspace_abs_path.join_within_root(package.locator.relpath)
+
+        qualifiers["vcs_url"] = str(package_abs_path)
```

Comment:
We need to take in consideration the `origin` of the repository, otherwise we end up with a local path that won't help locating the package on the internet:

```
pkg:npm/ansi-regex@6.0.1?vcs_url=/home/bpimente/Projects/cachito/test-repos/cachi2-yarn-berry/external-packages/ansi-regex#external-packages/ansi-regex
```

It should look like this:
```
pkg:npm/ansi-regex@6.0.1?vcs_url=git+https://github.com/cachito-testing/cachi2-yarn-berry@70515793108df42547d3320c7ea4cd6b6e505c46#external-packages/ansi-regex
```

Reply:
Fail it =/
https://github.com/containerbuildsystem/cachi2/blob/main/cachi2/core/scm.py#L49-L56

We should probably revisit this later, as I think it makes for bad user experience.